fv3core/decorators.py

```diff
@@ -394,7 +394,8 @@ def __call__(
         # Can optimize this by marking stencils that need these
         axis_offsets = fv3core.utils.axis_offsets(spec.grid, origin, domain)
 
-        regenerate_stencil = not self.built or global_config.get_rebuild()
+        self.rebuild = global_config.get_rebuild()
+        regenerate_stencil = not self.built or self.rebuild
```

Comment:
Is it important that line 397 be here instead of inside the `__init__` method?

Reply:
This also changes the internal state of the StencilWrapper in the __call__ method which is unintuitive to me.

Reply:
The issue is that the `@gtstencil` decorated stencils are loaded at import time before the variables are set in `global_config` so they need to be reset here (for now).